pkg/adapter/incoming.go

```diff
@@ -68,8 +76,20 @@ func (l *listener) detectIncoming(ctx context.Context, req *http.Request, payloa
 		return false, nil, fmt.Errorf("secret passed to the webhook is %s which does not match with the incoming webhook secret %s in %s", secretValue, querySecret, hook.Secret.Name)
 	}
 
-	if repo.Spec.GitProvider.Type == "" {
-		return false, nil, fmt.Errorf("repo %s has no git provider type, you need to specify one, ie: github", repository)
+	var isProviderTypeNotSet bool
+	if repo.Spec.GitProvider == nil {
+		isProviderTypeNotSet = true
+	} else if repo.Spec.GitProvider.Type == "" {
+		isProviderTypeNotSet = true
+	}
+	if isProviderTypeNotSet {
+		if err = l.getAndUpdateInstallationID(ctx, req, l.run, repo); err != nil {
```

Comment:
receiver function `l.getAndUpdateInstallationID` can access `l.run` itself, is there a reason to pass it from here?

Reply:
Refactored code and now it won't have receiver type